examples/plugins/src/file-size.plugin.ts

```diff
@@ -0,0 +1,212 @@
+    ...fileSizeAuditOutput,
+    score: scoreFilesizeAudit(issues.length, errorCount),
+    value: errorCount,
+    displayValue: displayValue(errorCount),
+  };
+
+  if (issues.length) {
+    fileSizeAuditOutput = {
+      ...fileSizeAuditOutput,
+      details: {
+        issues,
+      },
+    };
+  }
+
+  return [fileSizeAuditOutput];
+}
+
+export function scoreFilesizeAudit(issues: number, errors: number): number {
+  if (issues < errors) {
+    throw new Error(`issues: ${issues} cannot be less than errors ${errors}`);
+  }
+  issues = Math.max(issues, 0);
+  errors = Math.max(errors, 0);
+  return errors > 0 ? Math.abs((issues - errors) / issues) : 1;
+}
+
+export function displayValue(numberOfFiles: number): string {
+  return `${numberOfFiles} ${
+    numberOfFiles === 1 ? 'file' : pluralize('file')
+  } oversize`;
+}
+
+export async function fileSizeIssues(options: {
+  directory: string;
+  pattern?: string | RegExp;
+  budget?: number;
+}): Promise<Issue[]> {
+  const { directory, pattern, budget } = options;
+
+  const files = await readdir(directory);
+  const issuesPromises = files.map(async file => {
+    const filePath = join(directory, file);
+    const stats = await stat(filePath);
+
+    // depth first crawling
+    if (stats.isDirectory()) {
+      return fileSizeIssues({ directory: filePath, pattern, budget });
+    }
+
+    if (stats.isFile()) {
+      if (pattern !== undefined) {
+        if (new RegExp(pattern).test(file)) {
+          return assertFileSize(filePath, stats.size, budget);
+        }
+      } else {
+        return assertFileSize(filePath, stats.size, budget);
+      }
+    }
+
+    // flatMap will remove empty arrays
+    return [];
+  });
+
+  // Resolve all promises and flatten the array of issues
+  const issuesNestedArray = await Promise.all(issuesPromises);
+  return issuesNestedArray.flat();
+}
+
+export function infoMessage(filePath: string, size: number) {
+  return `File ${basename(filePath)} is OK. (size: ${formatBytes(size)})`;
+}
+
+export function errorMessage(filePath: string, size: number, budget: number) {
+  const sizeDifference = formatBytes(size - budget);
+  const byteSize = formatBytes(size);
+  const byteBudget = formatBytes(budget);
+  return `File ${basename(
+    filePath,
+  )} has ${byteSize}, this is ${sizeDifference} too big. (budget: ${byteBudget})`;
+}
+
+export function assertFileSize(
+  file: string,
+  size: number,
+  budget?: number,
+): Issue {
+  let severity: IssueSeverity = 'info';
+  // ensure size positive numbers
+  size = Math.max(size, 0);
+  let message = infoMessage(file, size);
+
+  if (budget !== undefined) {
+    // ensure budget is positive numbers
+    budget = Math.max(budget, 0);
+    // set severity to error if budget exceeded
+    if (budget < size) {
+      severity = 'error';
+      message = errorMessage(file, size, budget);
+    }
+  }
+  // return Issue
+  return {
+    message,
+    severity,
+    source: {
+      file,
```

Comment:
You can use the [`toUnixPath` function](https://github.com/code-pushup/cli/blob/main/packages/utils/src/lib/file-system.ts#L7-L18) from `utils`, this is what's [used by the ESLint plugin](https://github.com/code-pushup/cli/blob/main/packages/plugin-eslint/src/lib/runner/lint.ts#L19) to normalize file paths.

```suggestion
      file: toUnixPath(file, { toRelative: true }),
```

Then you wouldn't have the absolute path included:
![image](https://github.com/code-pushup/cli/assets/34691111/e87f7da9-4de3-499e-b1c3-9caa012acc3e)

Reply:
done